app/controllers/admin_users_sessions_controller.rb

```diff
@@ -24,4 +24,12 @@ def create
 
     redirect_to user_path(@user_to_login_as)
   end
+
+  def destroy
+    @admin_to_revert_to = User.find(session[:admin_id])
+
+    sign_in(@admin_to_revert_to)
+
+    redirect_to admin_general_index_path
```

Comment:
Might be nicer to redirect back to the `@user_to_login_as` user admin page, since that's where you "started" from to get to the login as mode.